src/main/java/com/mraof/minestuck/client/gui/toasts/GristToast.java

```diff
@@ -0,0 +1,165 @@
+package com.mraof.minestuck.client.gui.toasts;
+
+import com.mojang.blaze3d.systems.RenderSystem;
+import com.mojang.blaze3d.vertex.*;
+import com.mraof.minestuck.alchemy.GristType;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.components.toasts.Toast;
+import net.minecraft.client.gui.components.toasts.ToastComponent;
+import net.minecraft.client.renderer.GameRenderer;
+import net.minecraft.network.chat.TextComponent;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.server.level.ServerPlayer;
+import net.minecraft.util.Mth;
+import net.minecraftforge.api.distmarker.Dist;
+import net.minecraftforge.api.distmarker.OnlyIn;
+
+@OnlyIn(Dist.CLIENT)
+public class GristToast implements Toast
+{
+	ResourceLocation TEXTURE = new ResourceLocation("minestuck", "textures/gui/toasts.png");
+	
+	private static final long DISPLAY_TIME = 5000L;
+	private static final float SCALE_X = 0.6F;
+	private static final float SCALE_Y = 0.6F;
+	private GristType type;
+	private long difference;
+	private EnumSource source;
+	private boolean increase;
+	private long lastChanged;
+	private boolean changed;
+	
+	public enum EnumSource {
+		CLIENT,
+		SERVER,
+		SENDGRIST
+	}
```

Comment:
This enum is not used client-side only, so I suggest moving it elsewhere, such as to `GristHelper`